app/src/main/java/tttTDD/Java/Interfaces/Player.java

```diff
@@ -0,0 +1,6 @@
+package tttTDD.Java.Interfaces;
+
+public interface Player {
+    String getMarker();
+    boolean isComputer();
+}
```

Comment:
The player interface seems fairly anemic.

- Are there any fields that any implementation should always have?
- Should `Players` be able to _do_ anything, other than identify their marker and android character?

Reply:
It should be this lean since it's shared between the `Computer` and `Human` player. Anything more would imply that it knows about the other classes. One of the questions I brought up during IPM whether the computer needed to know about the available moves and if it did, would that mean it also has access to the board.

Reply:
> [Does] the computer [need] to know about the available moves and if it [does], would that mean it also has access to the board?

The answers to these questions flow from an earlier one, which is: what is the Computer (and relatedly the Player) actually supposed to _do_? In your current design and implementation, the answer is that Players do nothing at all, other than expose their (immutable) state to other objects. So, they also don't need to know or access anything external.

If instead you envisioned the Player class performing some action, then yes, it would likely reference your other classes, like Board.

Reply:
Yes. I'm sure there are other ways of pulling it off, but the most straightforward way of one object acting on another is for the first to call the second's public methods. That is distinct from the question of whether `Player` should be able to directly mutate the `Board`'s fields, which OOP discourages.